src/main/java/org/spongepowered/common/mixin/core/network/MixinNetHandlerPlayServer.java

```diff
@@ -239,4 +243,31 @@ public String dontNormalizeStringBecauseSomeMojangDevSucks(String input) {
         return input;
     }
 
+    @Inject(method = "processPlayer", at = @At(value = "FIELD", target = "net.minecraft.network.NetHandlerPlayServer.hasMoved:Z", ordinal = 2), cancellable = true)
+    public void proccesPlayerMoved(C03PacketPlayer packetIn, CallbackInfo ci){
+        if (packetIn.isMoving() || packetIn.getRotating()) {
+            Player player = (Player) playerEntity;
+            Vector3d fromrot = player.getRotation();
+            Location from = player.getLocation();
+            Vector3d torot = new Vector3d(packetIn.getYaw(), packetIn.getPitch(), 0);
+            Location to = new Location(player.getWorld(), packetIn.getPositionX(), packetIn.getPositionY(), packetIn.getPositionZ());
+
+            double delta = Math.pow(from.getX() - to.getX(), 2) + Math.pow(from.getX() - to.getY(), 2) + Math.pow(from.getX() - to.getZ(), 2);
```

Comment:
The delta is fine squared as long as the variables it is compared to also are squared. This saves a square root operation. I would strongly recommend calling it `deltaSquared` and having `1f / 256` written as `(1f / 16) * (1f / 16)` to make the constant clearer.